go/felix/usagerep/usagerep_test.go

```diff
@@ -60,4 +61,27 @@ var _ = Describe("Usagerep", func() {
 		Expect(q.Get("guid")).To(Equal("baddecaf"))
 		Expect(q.Get("cluster_type")).To(Equal("unknown"))
 	})
+	It("should delay at least 5 minutes", func() {
+		Expect(calculateInitialDelay(0)).To(BeNumerically(">=", 5*time.Minute))
+		Expect(calculateInitialDelay(1)).To(BeNumerically(">=", 5*time.Minute))
+		Expect(calculateInitialDelay(1000)).To(BeNumerically(">=", 5*time.Minute))
+	})
+	It("should have a random component", func() {
+		Expect(calculateInitialDelay(1000000)).ToNot(Equal(calculateInitialDelay(1000000)))
```

Comment:
This test - with 1 million hosts - seems unrealistic.  Is the number so large because you're trying to reduce the likelihood of a possible collision?  If so, I think a better test would be to call calculateInitialDelay(50) 10 times, and expect that at least one of the values is different.